### PR TITLE
Introduces nightly artifact builds

### LINKED ISSUE #1677: Introduce nightly artifact production
## Abstract

As part of the scheduled test cron job on GitHub Actions, we need to introduce nightly artifact builds based on the release CI configuration.

### Tasks
- [ ] Linux AppImage
- [ ] OSX dmg

### PR DESCRIPTION
**Pull request by kelsos**
Closes #1677

### CODE REVIEW

**Comment by kelsos** [comment]
I did a successful test run here https://github.com/kelsos/rotki/actions/runs/343668546

File: .github/workflows/rotki_nightly.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
       - name: Download Geth
         run: |
           .travis/download_geth.sh ${{ matrix.os }}
-          echo "::add-path::$HOME/.bin"
+          echo "PATH=$HOME/.bin:$PATH" >> $GITHUB_ENV
```
Comment (kelsos) [inline .github/workflows/rotki_nightly.yml:21]: The `add-path` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

**Review by LefterisJP — APPROVED** [review]
LGTM. Can't really test. You know the actions best.

But 2 questions:

1. Could this affect our normal building and surprise us  down the line?
2. Does it consume some kind of credits running it every night? If so perhaps we could run them on demand? Or am I overthinking it?

**Comment by kelsos** [comment]
> LGTM. Can't really test. You know the actions best.
> 
> But 2 questions:
> 
>     1. Could this affect our normal building and surprise us  down the line?
> 
Nope not really, it is a separate workflow with some slight changes compared to the tag building

>     2. Does it consume some kind of credits running it every night? If so perhaps we could run them on demand? Or am I overthinking it?

I would assume no, last time we checked github had only actual limits for private repositories on the free plan, but I am assuming some non abuse policy without explicit limits when running on a public open repo.

**Pull request merged**